### PR TITLE
New PR on hotfix branch: Simplify registering tools

### PR DESCRIPTION
**Pull request by bethsheets**
New PR replacing #187 
- I removed a lot of information to try to simplify the getting started. If a user is publishing a tool for the first time, I don't think they need to revisit our discussion of why we changed the tool definition.
- I removed the section about searching for other tools after registering your own tool. We should separate these user personas (tool developer vs. someone searching for a tool to reuse "end user").

### CODE REVIEW

**Comment by aofarrel** [comment]
The CircleCI failure lead to a new discovery. The old version said:

> :doc:`in our Dockstore GitHub App </getting-started/github-apps/github-apps> doc`

And it turns out that it was attempting to interpret " doc" as part of the URL. It looks like that once you start the URL, you must cap it with >\` before typing anything else.

File: docs/getting-started/dockstore-tools.rst [hunk 1]
```diff
@@ -13,46 +13,39 @@ Tutorial Goals
...
+This document assumes you already have a tool ready to register. 
 
-If you followed :doc:`Getting Started With CWL </getting-started/getting-started-with-cwl>` to create a CWL tool, you will now have your ``Dockerfile`` and ``Dockstore.cwl`` in GitHub, have setup Quay.io to automatically build your Docker image, and have linked your accounts to Dockstore. This will allow you to use any of our registration methods, including legacy methods. Of course, you can always follow along with your own unique tools too, which may or may not be associated with an autobuilding Docker image.
+If you followed :doc:`Getting Started With CWL </getting-started/getting-started-with-cwl>` to create a CWL tool, you will now have your ``Dockerfile`` and ``Dockstore.cwl`` in GitHub, have setup Quay.io to automatically build your Docker image, and have linked your accounts to Dockstore. This will allow you to use any of our registration methods, including legacy methods. You can also follow this tutorial if your tool is not associated with an autobuilding Docker image.
```
Comment (aofarrel) [inline docs/getting-started/dockstore-tools.rst:18]: Getting started with CWL has since changed; the old tutorial which included a Dockerfile and autobuilding is now gone. (Well, strictly speaking, [moved](https://docs.dockstore.org/en/stable/advanced-topics/legacy/legacy-getting-started-with-cwl.html).) I propose changing this paragraph to the following.


> If you followed :doc:\`Getting Started With CWL </getting-started/getting-started-with-cwl>\` to create a CWL tool, you will now have a \`\`Dockstore.cwl\`\` in GitHub and have linked your accounts to Dockstore. This will allow you to use the Dockstore GitHub App to register your tool. Of course, you can always follow along with your own unique tools too. Regardless of how you made your tool, this tutorial will assume you are using CWL. For all other languages, :doc:\`please see our documentation on workflows instead </getting-started/dockstore-workflows>\`.

File: docs/getting-started/dockstore-tools.rst [hunk 2]
```diff
@@ -13,46 +13,39 @@ Tutorial Goals
...
 
-Regardless of how you made your tool, this tutorial will assume you are using CWL. For all other languages, :doc:`please see our documentation on workflows instead </getting-started/dockstore-workflows>`.
-
-.. note:: The WDL community does not explicitly differentiate :doc:`tools versus workflows </getting-started/intro-to-dockstore-tools-and-workflows>`. However, Dockstore allows the registration of "WDL tools" using legacy registration methods. We encourage people to :doc:`register WDLs as workflows </getting-started/dockstore-workflows>` instead.
+.. note:: Dockstore allows the registration of "WDL tools" using legacy registration methods. The WDL community does not explicitly differentiate :doc:`tools versus workflows </getting-started/intro-to-dockstore-tools-and-workflows>`. We encourage people to :doc:`register WDLs as workflows </getting-started/dockstore-workflows>` instead.
```
Comment (aofarrel) [inline docs/getting-started/dockstore-tools.rst:20]: This is a nitpick, but I feel that a qualifier is needed between the statements "you can register WDL tools" and "WDL doesn't have a concept of tools" to underline that the Dockstore and the WDL community have opposing definitions. If a qualifier is not given, then the contradiction between "you can register WDL tools" and "WDL doesn't have a concept of tools" is not being acknowledged, which may lead the user astray.

Basically I'm picturing Johnny User reading this and saying "You said you can register tools on Dockstore, then you say the WDL community says there is no difference between tools and workflows. That must mean either:
A) There is no meaningful difference between WDL tools and WDL workflows on Dockstore [false]
B) You made a typo and meant that the WDL community **does** make a distinction between them [false]"

File: docs/getting-started/dockstore-tools.rst [hunk 3]
```diff
@@ -13,46 +13,39 @@ Tutorial Goals
...
 .. include:: /getting-started/github-apps/note--registration.rst
 
-There are a variety of ways to get your tools into Dockstore. Users can either use GitHub App registration or our legacy registration methods.
-GitHub App registration is the recommended way to register all new tools on Dockstore. GitHub App tools and tools registered using our other methods (legacy tools) are
-very different from one another. Use the following questions to determine which method to use:
+We recommend users register their tool with our GitHub App to benefit from GitHub's features and automatic syncing. If you do not wish to use GitHub, we do support other methods of registration in our legacy registration methods. Use the following questions to determine which registration method works for you:
 
 .. include:: /getting-started/how-to-register-work.rst
 
-If you must use the legacy tool registration methods, then you may want to read :doc:`Dockstore Tools Overhaul </../advanced-topics/dockstore-tools-overhaul>` before continuing
```
Comment (aofarrel) [inline docs/getting-started/dockstore-tools.rst:34]: I agree with removing this part out of this section, but I think there needs to be a link to Dockstore Tools Overhaul somewhere in the section directly under the legacy tools registration header on line 74.

File: docs/getting-started/dockstore-tools.rst [hunk 4]
```diff
@@ -92,23 +85,16 @@ Sharing Your Tool
...
+synced with GitHub, Bitbucket, or GitLab, or hosted directly in Dockstore, you are now ready to share your
 tool with the public! Assuming that your tool has at least one valid
 version, you can publish your tool for everyone to use. Simply select the
-tool on the ``/my-tool`` page and click publish.
-
-Find Other Tools
-----------------
-
-You can find tools on the Dockstore website or also through the
-``dockstore tool search`` command line option.
```
Comment (aofarrel) [inline docs/getting-started/dockstore-tools.rst:104]: It is worth mentioning the CLI's search feature appears to be documented nowhere else, but I made a ticket for that (https://github.com/dockstore/dockstore/issues/4835) and would be alright with removing this section.

File: docs/getting-started/dockstore-tools.rst [hunk 5]
```diff
@@ -92,23 +85,16 @@ Sharing Your Tool
...
+tool on the ``/my-tool`` page and click `publish`.
 
 Next Steps
 ----------
 
-You can follow this basic pattern for each of your Docker-based tools.
```
Comment (aofarrel) [inline docs/getting-started/dockstore-tools.rst:109]: Thanks for catching this!

**Review by aofarrel — CHANGES_REQUESTED** [review]
This is swell. Just a few changes requested.

File: docs/getting-started/dockstore-tools.rst (hunk 4 above)
Comment (aofarrel) [reply docs/getting-started/dockstore-tools.rst:104]: (Not requesting any changes here, just making note that this functionality is not mentioned elsewhere, in case that influences whether or not you decide to keep the deletion -- totally fine with either outcome.)

File: docs/getting-started/dockstore-tools.rst (hunk 1 above)
Comment (aofarrel) [reply docs/getting-started/dockstore-tools.rst:18]: On a second reading, I think the third and forth sentence that I wrote just above could be improved... feel free to play with that paragraph until it sounds right, I just think we need to acknowledge that the current Getting Started With CWL does not involve Dockerfiles.

File: docs/getting-started/dockstore-tools.rst (hunk 3 above)
Comment (bethsheets) [reply docs/getting-started/dockstore-tools.rst:34]: I disagree- If getting started is aimed at new users, why do they need to read about what we previously did? I don't think returning users will be heavy users of this page.

Comment (aofarrel) [reply docs/getting-started/dockstore-tools.rst:34]: The problem is that legacy registration creates legacy tools, which are very different from our new definition of a tool. So anyone using legacy registration needs to know the history.

Comment (bethsheets) [reply docs/getting-started/dockstore-tools.rst:34]: Ok, thanks for clarifying. I added a link to line 80. I think adding there helps keep it from disrupting the main tutorial flow.

**Review by aofarrel — APPROVED** [review]

**Review by denis-yuen — APPROVED** [review]
